src/main/java/org/broadinstitute/hellbender/engine/MultiFeatureWalker.java

```diff
@@ -0,0 +1,264 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.tribble.Feature;
+import htsjdk.variant.vcf.VCFHeader;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.tools.sv.SVFeaturesHeader;
+
+import java.util.*;
+
+/**
+ * A MultiFeatureWalker is a tool that presents one {@link Feature} at a time in sorted order from
+ * multiple sources of Features.  The input files for each feature must be sorted by locus.
+ *
+ * To use this walker you need only implement the abstract
+ * {@link #apply(F, Object, ReadsContext, ReferenceContext)} method in a class that declares
+ * a collection of FeatureInputs as an argument.
+ * You may optionally implement {@link #onTraversalStart()}, {@link #onTraversalSuccess()},
+ * and/or {@link #closeTool()}.
+ */
+public abstract class MultiFeatureWalker<F extends Feature> extends WalkerBase {
+
+    SAMSequenceDictionary dictionary;
+    final Set<String> samples = new TreeSet<>();
```

Comment:
Since there are accessor methods for these, can they be `private`? And relatedly, should the Set of samples be wrapped in a `Collections.unmodifiableSet()` after initialization to prevent downstream modification?